app/modules/pagination/actions/pagination-fetch-page.js

```diff
@@ -0,0 +1,24 @@
+import {paginationReceivePage} from "./pagination-receive-page"
+import {paginationSetNextPage} from "./pagination-set-next-page"
+import {submissionCreate} from "../../submissions/actions/submission-create"
+import LinkHeader from "http-link-header"
+
+export const fetchPage = (repoURL, page) => {
+  return dispatch => {
+    return fetch(`${repoURL}?page=${page}`, {
+      credentials: "include"
+    }).then(response => {
+      dispatch(paginationSetNextPage(null)) // Set next page to null, unless we get header
```

Comment:
:art: the comment here suggests we can structure this code to avoid unnecessary `dispatch`-ing, but maybe I'm missing some context:

```js
let nextPage = null

if (response.headers.get("Link")) {
  var link = LinkHeader.parse(response.headers.get("Link"))
  if (link.has("rel", "next") && link.get("rel", "next").length > 0) {
   nextPage = link.get("rel", "next")[0].params.page)
  }
}

dispatch(paginationSetNextPage(nextPage))
return response.json()
```

Reply:
This is a much better way to structure this code since it avoids the unnecessary dispatch, thanks! Updated the code to reflect this change